File: L2.cpp
#include <bits/stdc++.h>
#include <iterator>
using namespace std;

// interface .h
class bolas
{
private:
	int *val;
public:
	bolas();
	~bolas();
	void init(int v);
	void setVal(int v, int w);
	int getVal(int v);
	void sort();
	
};



class jogador
{
private:
	int *val;
public:
	jogador();
	~jogador();
	void init(int w);
	int getScore(bolas b, int k);
	
};


// implementação .cpp

bolas::bolas(){}

bolas::~bolas(){}

void bolas::init(int w){
	this->val = (int *)calloc(w, sizeof(int));
}

void bolas::setVal(int pos, int v){
	this->val[pos] = v;
}

int bolas::getVal(int pos){ return this->val[pos]; }

void bolas::sort(){
	std::sort(this->val, (this->val) + 10);	
}




jogador::jogador(){}

jogador::~jogador(){}

void jogador::init(int w){
	this->val = (int *)calloc(w, sizeof(int));
}

int jogador::getScore(bolas b, int k){
	for (int i = 0; i < k; ++i){
		/* code */
	}
}
// source.cpp 

int main(){

	bolas b;
	int n, m, k, p, a, v;

	while( scanf("%d %d %d %d %d", &n, &m, &k, &p, &a) != EOF){
		b.init(n);
		for (int i = 0; i < n; ++i){
			cin >> v;
			b.setVal(i, v);
			cout << "bola " << i << ": " << b.getVal(i) << endl;
		}
		for (int i = 0; i < p; ++i){
			for (int j = 0; j < k; ++j){
				
			}
		}
		b.sort();
		cout << endl;
		for (int i = 0; i < n; ++i){
			cout << "bola " << i << ": " << b.getVal(i) << endl;
		}

	}

	return 0;
}

File: L2.py
class bolas():
	def __init__(self):
		self.__vet = []

	def insertBall(self, w):
		self.__vet.append(w)

	def getVal(self, v):
		return self.__vet[v]

	def sort(self):
		self.__vet.sort()




class jogador():
	def __init__(self):
		self.__score = 0

	def getScore(self):
		return self.__score
	
	def setScore(self, n, v, b):
		l = 0; r = n-1;

		while l <= r:
			m = int((l+r)/2)
			if v == b.getVal(m):
				return m

			elif v < b.getVal(m):
				r = m-1

			elif v > b.getVal(m):
				l = m+1

	def addScore(self, v):
		self.__score += v




def main():
	caso = 0
	while(True):
		try:
			n = input()
			if n == "":
				n = input()	
		except EOFError:
			break

		m = int(n.split()[1])
		k = int(n.split()[2])
		p = int(n.split()[3])
		a = int(n.split()[4])
		n = int(n.split()[0])

		print("caso {}:".format(caso), end="")
		caso += 1

		b = bolas()
		j = []

		for x in range(0,n):
			aux = int(input())
			b.insertBall(aux)

		b.sort()

		for i in range(0,p):
			for t in range(0,k):
				aux = int(input())
				j.append(jogador())
				j[i].addScore(j[i].setScore(n, aux, b))

			#print("jogador {}: {}".format(i, j[i].getScore()))	
			if j[i].getScore() == a:
				print (" {}".format(i), end ="")

		print(""); print("");



if __name__ == '__main__':
	main()